src/generic/C1_helper.h

```diff
@@ -0,0 +1,1092 @@
+ /// (or for large-amplitude problems, e.g. Koiter Steigman, where all three
+ /// displacement components need to be C1. Broken function will shout if it's
+ /// evaluated.
+ virtual double dfdn(const double& zeta)
+  {
+   std::stringstream error_message;
+   error_message
+    << "You'll have to implement the function that\n"
+    << "specifies the normal derivative if you want to use it!\n"
+    << "This broken virtual function was probably called \n"
+    << "when you tried to implement clamping bcs in a plate problem.\n"
+    << std::endl;
+   throw OomphLibError(error_message.str(),
+                       OOMPH_CURRENT_FUNCTION,
+                       OOMPH_EXCEPTION_LOCATION);
+
+  }
+ 
+ /// Virtual function to specify the derivative of the function (typically a
+ /// displacement component) w.r.t zeta, the 1D coordinate that parametrises the
+ /// boundary. Defaults to fd evaluation. Overload with your own version if
+ /// you prefer
+ double dfdzeta(const double& zeta)
+  {
+   return (f(zeta+FD_step)-f(zeta))/FD_step;
+  }
+ 
+
+ /// Virtual function to specify the derivative of the normal derivatives 
+ /// w.r.t zeta , the 1D coordinate that parametrises the boundary. Defaults to
+ /// fd evaluation. Overload with your own version if you prefer.
+ double d2fdndzeta(const double& zeta)
+  {
+   return (dfdn(zeta+FD_step)-dfdn(zeta))/FD_step;
+  }
+
+ /// Virtual function to specify the second derivative of the function
+ /// w.r.t zeta , the 1D coordinate that parametrises the boundary. Defaults to
+ /// fd evaluation. Overload with your own version if you prefer.
+ double d2fdzeta2(const double& zeta)
+  {
+   return (f(zeta+FD_step)-2.0*f(zeta)+f(zeta-FD_step))/(FD_step*FD_step);
+  }
+
+
+private:
+
+ /// FD step
+ double FD_step;
+
+};
+
+
+ ////////////////////////////////////////////////////////////////////////
+ ////////////////////////////////////////////////////////////////////////
+ ////////////////////////////////////////////////////////////////////////
+ 
+
+
+ 
+ //===start of rotation helper class=========================================
+ /// Helper class to contain all the rotation information in the element.
+ //==========================================================================
+ class RotatedBoundaryHelper
+  {
+  public:
+   
+   /// Constructor: just initialise the member data to their defaults (zeros)
+   /// hierher Aidan: what's the parent element?
+   RotatedBoundaryHelper(FiniteElement* const& parent_element_pt)
+    : Parent_element_pt(parent_element_pt),
+      Nnode(Parent_element_pt->nvertex_node()),
+      Boundary_coordinate_of_node(3, 0.0),
+      Nodal_boundary_parametrisation_pt(3, 0),
+      Rotation_matrix_at_node(3, DenseMatrix<double>(6, 6, 0.0))
+    {
+    }
+   
+   /// Destructor
+   ~RotatedBoundaryHelper() {}
+
+   /// Return C1Curviline associated with the boundary
+   /// parametrisatino that given local node is on
+   /// hierher Aidan: please confirm
```

Comment:
Confirm what? This should return the parametrisation pt for the curve the node is on.

Reply:
Aidan confirmed.